### PR TITLE
V0.x dev

### PR DESCRIPTION
**Pull request by spicysomtam**
Added 3 files to allow folks to build a docker image of pano v0.x. Doesn`t affect the code in the release. The docker build does a git clone, so it will pull the latest down from github and drop that into the Docker image. Hopefully the readme will allow you to try it out and get some experience with docker (which is rather awesome!). Alastair

### CODE REVIEW

**Comment by sheijmans** [comment]
It would be nice to move the config files (panopuppet.conf & comfig.yaml) and the sqllite db outside the container.

**Comment by sheijmans** [comment]
For the build/proxy option it is better to do it like;
`docker build -t panopuppet:v0.x  --build-arg http_proxy=http://proxy.company.co.uk:8080 --build-arg https_proxy=https://proxy.company.co.uk:8080 .`

https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables-build-arg
`Also, these values don’t persist in the intermediate or final images like ENV values do.`

**Comment by spicysomtam** [comment]
That option isn't available in docker 1.7.1; that is the latest version for centos 6. Because of the 2.6.32 kernel on el6 being less capable, there will not be any later versions of docker for centos 6. See this: https://github.com/docker/docker/issues/14365. Also you need to install docker-io instead of docker on centos 6 (for pano v0.x).

I wrote this earlier, but got distracted and didn't hit the Comment button: 

The config.yaml is copied to the image at image build time, so the intention there is that its already configured and ready to go.

Regarding putting stuff outside the container: this is more of a container deployment question after the image is built. Suppose it depends on what you want. For me I just wanted a container that I can spin up that will connect to our pdb, so copying the config.yaml in at image build time was sufficient. This also follows the idea that containers should be light and not have complex setups. Users authenicate via the ldap settings in config.yaml so that works fine. If you are concerned about making a backup of the sqllite db, etc, you could have say a cronjob doing a docker cp from time to time to pull those files out of the container. If you want the /srv/repo/panopuppet dir to be outside the container, it will need to be copied out once the container is deployed and then redeployed with -v to bind some mount/s. I think in the latest version of pano there is a SQLITE_DIR setting in the config.yaml, so that is another way of tweaking the config. So its really up to what you want. I wanted to keep it simple.

In a nutshell I just provided the docker image build part and some docs. Regarding the container deployment from the image, this is really up to the user. I think if we considered all the options it would become complicated. I can certainly update the DOCKER.md docs to include the senarios above, so users can choose what they want? Maybe a _Managing data that changes_ section?

**Comment by propyless** [comment]
Given the thought you have put into this and also regarding an addition to the 0.x release were the instructions are for centos 6.
People are given a boilerplate template with which they can do whatever they want.

Merged.

**Comment by spicysomtam** [comment]
Let me know if you want me to add some more documentation to the DOCKER.md. You can never have to much documentation :)
